### PR TITLE
✨ Mobile | V3 Redeem page UI

### PR DESCRIPTION
**Pull request by zacharykeeping**
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

Part of #560 and #736 

> 2. What was changed?

Implements the UI for the Redeem page and part of the redemption process. Handling the redemption process after confirmation is still yet to be done (clicking Confirm currently does nothing).

Also implements some design tweaks to better match the V3 mockups and handle varying text sizes, and allows for placeholder icons in our lists.

**Figure: Redeem page**

**Figure: Redemption popup part 1**

**Figure: Redemption popup part 2**

> 3. Did you do pair or mob programming?

No

### CODE REVIEW

File: src/MobileUI/Controls/CarouselItem.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+public partial class CarouselItem : Border
+{
+    public string CarouselImage
+    {
+        get => (string)GetValue(CarouselImageProperty);
+        set => SetValue(CarouselImageProperty, value);
+    }
+
+    public static readonly BindableProperty CarouselImageProperty =
```
Comment (matt-goldman) [inline src/MobileUI/Controls/CarouselItem.xaml.cs:15]: FYI we could probably use this to make it easier: https://github.com/rrmanzano/maui-bindableproperty-generator

I've lodged this issue for it: https://github.com/SSWConsulting/SSW.Rewards.Mobile/issues/744

File: src/MobileUI/PopupPages/RedeemReward.xaml [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+                    
+                    <Label Text="Postcode"
+                           FontSize="18"
+                           TextColor="{StaticResource Gray300}"/>
+                    <Entry Text="{Binding AddressPostcode}"
+                           Margin="0,0,0,8"/>
+                    
+                    <FlexLayout BindableLayout.ItemsSource="{Binding States}"
+                                Wrap="Wrap">
+                        <BindableLayout.ItemTemplate>
```
Comment (matt-goldman) [inline src/MobileUI/PopupPages/RedeemReward.xaml:182]: This UI is a bit clunky. Can we ask the designers to review this? We might be able to improve it in a future iteration.

**Review by matt-goldman — APPROVED** [review]
Looks good to me! 🚀

File: src/MobileUI/Controls/CarouselItem.xaml.cs (hunk 1 above)
Comment (zacharykeeping) [reply src/MobileUI/Controls/CarouselItem.xaml.cs:15]: Oh nice, I was thinking there had to be a better way to go about this... This would be way cleaner.

**Pull request merged**